modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/granulecollector/ReprojectingSubmosaicProducer.java

```diff
@@ -158,13 +206,52 @@ public CRSBoundMosaicProducer(RasterLayerResponse rasterLayerResponse, boolean d
 
         @Override
         public boolean accept(GranuleDescriptor granuleDescriptor) {
+            //make sure the CRSs match
+            boolean shouldAccept = false;
 
+            //need to check that the granule matches CRS
+            try {
+                CoordinateReferenceSystem  granuleCRS = getCRS(
+                    getGranuleCRSCode(granuleDescriptor));
+                shouldAccept = CRS.equalsIgnoreMetadata(granuleCRS, this.crs);
+            } catch (FactoryException e) {
+                throw new IllegalArgumentException("Unable to create granule CRS from CRS code. "
+                    + "Cannot proceed with mosaicking.", e);
+            }
+
+            //we need to see if the incoming granule matches the "template" feature this producer
+            //was initialized with.
+            if (sortBy != null) {
+                SimpleFeature granuleFeature = granuleDescriptor.getOriginator();
+                SimpleFeature templateFeature = templateDescriptor.getOriginator();
+                for (SortBy sortKey : sortBy) {
+                    Object incomingAttribute = granuleFeature
```

Comment:
I'm not sure I understand why this is necessary. I though ensuring uniform CRS in the producer was enough. E.g., if I have all features in the same CRS but I'm sorting on date, do I really need to roll each feature in its own sub-mosaicker?